pybikes/clujbike.py

```diff
@@ -19,70 +20,33 @@ def __init__(self, tag, feed_url, meta):
+        scraper = scraper or PyBikesScraper()
 
-        stations = []
+        raw = scraper.request(self.feed_url)
 
-        post_data = {
-            'sort' : '',
-            'group' : '',
-            'filter' : '',
-            'StationName' : '',
-            'Address' : '',
-            'OcuppiedSpots' : '0',
-            'EmptySpots' : '0',
-            'MaximumNumberOfBikes' : '0',
-            'LastSyncDate' : '',
-            'IdStatus' : '0',
-            'Status' : '',
-            'StatusType' : '',
-            'Latitude' : '0',
-            'Longitude' : '0',
-            'IsValid' : 'true',
-            'CustomIsValid' : 'false',
-            'Id' : '0',
-        }
+        data = re.search(r'var stations = (\[.*?\]);', raw).group(1)
+
+        stations = []
 
-        data = json.loads(
-            scraper.request(self.feed_url, 'POST', post_data, None)
-        )
-        # Each station is
-        # {
-        #     "StationName":"Biblioteca Centrala",
-        #     "Address":"Biblioteca Județeană Octavian Goga",
-        #     "OcuppiedSpots":0,
-        #     "EmptySpots":20,
-        #     "MaximumNumberOfBikes":20,
-        #     "LastSyncDate":"01/08/2016 12:26",
-        #     "IdStatus":1,
-        #     "Status":"Functional",
-        #     "StatusType":"Subpopulated",
-        #     "Latitude":46.777037,
-        #     "Longitude":23.615109,
-        #     "IsValid":true,
-        #     "CustomIsValid":false,
-        #     "Notifies":[],
-        #     "Id":85,
-        # }
-        for item in data['Data']:
-            name = item['StationName']
-            latitude = float(item['Latitude'])
-            longitude = float(item['Longitude'])
-
-            if float(latitude) == 0.0 or float(longitude) == 0.0:
-                continue
-
-            bikes = int(item['OcuppiedSpots'])
-            free = int(item['EmptySpots'])
-            status = 'offline' if item['StatusType'] == 'Offline' else 'online'
-            extra = {
-                'slots': item['MaximumNumberOfBikes'],
-                'address': item['Address'],
-                'status': status
-            }
-            station = BikeShareStation(name, latitude, longitude,
-                                       bikes, free, extra)
-            stations.append(station)
+        for station in json.loads(data):
+            stations.append(ClujBikeStation(station))
 
         self.stations = stations
+
+
+class ClujBikeStation(BikeShareStation):
+    def __init__(self, data):
+        super(ClujBikeStation, self).__init__()
+
+        self.name = data['StationName']
+        self.latitude = float(data['Latitude'])
+        self.longitude = float(data['Longitude'])
+
+        self.bikes = int(data['OcuppiedSpots'])
+        self.free = int(data['EmptyDoors'])
+        self.status = 'offline' if data['Status'] == 'Offline' else 'online'
+        self.extra = {
+            'slots': self.bikes + self.free,
```

Comment:
Not sure we can trust to properly represent the number of slots that stations have. I see there are many stations with 19 slots. I would go with removing it unless it's constantly accurate.

Reply:
Sure, just thought it could be helpful. Removed!